src/api/files/service.test.js

```diff
@@ -166,39 +206,268 @@ describe('Files service', () => {
 
+    it('should throw 410 Gone if file is missing', async () => {
+      const dummyData = {
+        ...successfulFile,
+        s3Key: 'dummy',
+        s3Bucket: 'dummy',
+        retrievalKey: 'test'
+      }
+
+      jest.mocked(verify).mockResolvedValueOnce(true)
+      jest.mocked(repository.getByFileId).mockResolvedValueOnce(dummyData)
+      s3Mock.on(HeadObjectCommand).rejectsOnce(
+        new NotFound({
+          message: 'Not found',
+          $metadata: {}
+        })
+      )
+
+      await expect(getPresignedLink('123-456-789', 'dummy')).rejects.toThrow(
+        Boom.resourceGone()
+      )
+    })
+
     it('should fail if the retrieval key does not match', async () => {
       const dummyData = {
         ...successfulFile,
         _id: new ObjectId(),
         retrievalKey: 'test'
       }
 
-      jest.mocked(verify).mockResolvedValue(false)
-      jest.mocked(repository.getByFileId).mockResolvedValue(dummyData)
+      jest.mocked(verify).mockResolvedValueOnce(false)
+      jest.mocked(repository.getByFileId).mockResolvedValueOnce(dummyData)
 
-      await expect(getPresignedLink('123-456-789', 'test')).rejects.toEqual(
+      await expect(getPresignedLink('123-456-789', 'test')).rejects.toThrow(
         Boom.forbidden('Retrieval key does not match')
       )
     })
   })
+
+  describe('persistFile', () => {
+    const newRetrievalKey = 'newKey'
+
+    beforeEach(() => {
+      s3Mock.reset()
+    })
+
+    it('should move the file from staging to loaded and delete the old file', async () => {
+      /** @type {FormFileUploadStatus} */
+      const dummyData = {
+        ...successfulFile,
+        s3Key: 'staging/dummy-file-123.txt',
+        retrievalKey: 'test'
+      }
+
+      const expectedNewKey = 'loaded/dummy-file-123.txt'
+
+      jest.mocked(verify).mockResolvedValueOnce(true)
+      jest.mocked(hash).mockResolvedValueOnce('newKeyHash')
+      jest.mocked(repository.getByFileId).mockResolvedValueOnce(dummyData)
+
+      await persistFile(
+        dummyData.fileId,
+        dummyData.retrievalKey,
+        newRetrievalKey
+      )
+
+      expect(hash).toHaveBeenCalledWith(newRetrievalKey)
+      expect(repository.updateRetrievalKey).toHaveBeenCalledWith(
+        dummyData.fileId,
+        'newKeyHash'
+      )
+
+      expect(repository.updateS3Key).toHaveBeenCalledWith(
+        successfulFile.fileId,
+        expectedNewKey
+      )
+
+      expect(s3Mock).toHaveReceivedCommandWith(CopyObjectCommand, {
+        Bucket: successfulFile.s3Bucket,
+        Key: expectedNewKey,
+        CopySource: 'dummy-bucket/staging/dummy-file-123.txt'
+      })
+
+      expect(repository.updateS3Key).toHaveBeenCalledWith(
+        successfulFile.fileId,
+        expectedNewKey
+      )
+
+      expect(s3Mock).toHaveReceivedCommandWith(DeleteObjectCommand, {
+        Bucket: successfulFile.s3Bucket,
+        Key: dummyData.s3Key
+      })
+    })
+
+    it("should fail if the retrieval key doesn't match", async () => {
```

Comment:
Is Prettier running?

```suggestion
    it('should fail if the retrieval key doesn't match', async () => {
```

Reply:
Yeah. Prettier uses double quotes when the string contains an apostrophe, rather than requiring escaping it.

Reply:
Oh yeah didn't see it!